coding/python_style_guide.rst

```diff
@@ -12,648 +13,671 @@ To request changes to these standards, please file an :ref:`RFC <decision-making
 
-.. _style-guide-py-4-2:
+3. Whitespace
+=============
 
-ASCII Encoding MUST be used for new code
-----------------------------------------
+Follow the `PEP 8 whitespace style guidelines <https://www.python.org/dev/peps/pep-0008/#id26>`_, with the following adjustments.
 
-- Always use ASCII for new python code.
+.. _style-guide-py-minimal-parens:
 
-- **Do not** include a coding comment (as described in  :pep:`263`) for ASCII files.
+The minimum number of parenthesis needed for correctness and readability SHOULD be used
+---------------------------------------------------------------------------------------
 
-- Existing code using Latin-1 encoding (a.k.a. ISO-8859-1) is acceptable so long as it has a proper coding comment. All other code must be converted to ASCII or Latin-1 except for 3rd party packages used "as is."
+Yes:
 
-.. _style-guide-py-4-3:
+.. code-block:: py
 
-DM specified code order SHOULD be followed
-------------------------------------------
+   a = b(self.config.nSigmaToGrow*sigma + 0.5)
 
-Within a module, follow the order: 
+Less readable:
 
-1. Shebang line (``#!``), only for executable scripts
-2. Module-level comments
-3. Module-level docstring
-4. Imports
-5. ``__all__`` statement, if any
-6. Module variables (names start with underscore)
-7. Module functions and classes (names start with underscore)
-8. Public variables
-9. Public functions and classes
-10. Optional test suites
+.. code-block:: py
 
-.. _style-guide-py-4-4:
+   a = b((self.config.nSigmaToGrow*sigma) + 0.5)
+ 
+.. _style-guide-py-operator-whitespace:
 
-Imports SHOULD be grouped, in order: standard lib, 3rd party lib, local lib
----------------------------------------------------------------------------
+Binary operators SHOULD be surrounded by a single space except for [``*``, ``/``, ``**``, ``//``, ``%``\ ]
+----------------------------------------------------------------------------------------------------------
+ 
+Always surround these binary operators with a single space on either side; this helps the user see where one token ends and another begins:
 
-Imports should be grouped in the following order, with each group separated by a blank line:
+- assignment (``=``),
+- augmented assignment (``+=``, ``-=``, etc.),
+- comparisons (``==``, ``<``, ``>``, ``!=``, ``<>``, ``<=``, ``>=``, ``in``, ``not in``, ``is``, ``is not``),
+- Booleans (``and``, ``or``, ``not``).
 
-1. standard library imports
-2. related third party imports
-3. local application/library specific imports
+Use spaces around these arithmetic operators:
 
-When importing a class from a class-containing module,
-it's usually okay to do this:
+- addition (``+``),
+- subtraction (``-``)
 
-.. code-block:: py
+Never surround these binary arithmetic operators with whitespace:
 
-   from myclass import MyClass
-   from foo.bar.yourclass import YourClass
+- multiplication (``*``),
+- division (``/``),
+- exponentiation (``**``),
+- floor division (``//``),
+- modulus (``%``).
```

Comment:
I believe that we do want spaces around `%` when it is used for string formatting (as opposed to as a math operator).